lib/get-transitive-data.js

```diff
@@ -439,42 +452,58 @@ function clusterDissimilarity (c1, c2, data) {
         const s1 = j1.segments[segment]
         const s2 = j2.segments[segment]
 
-        // if one has a walk segment where the other has a transit segment, that's a problem
-        // and these are not comparable
+        // if one has a walk segment where the other has a transit segment these
+        // are not comparable
         if (s1.type !== s2.type) return Infinity
 
-        // Don't look at walk segments, we only cluster based on the stop positions which we get from transit segments
+        // Only cluster  on the stop positions which we get from transit segments
         if (s1.type !== 'WALK') {
-          ret = Math.max(ret, segmentDissimilarity(s1, s2, data))
+          dissimilarity = Math.max(
+            dissimilarity,
+            segmentDissimilarity(s1, s2, {patterns, stops})
+          )
 
           // no point in continuing, these won't be merged
-          if (ret > MAXIMUM_DISSIMILARITY) return Infinity
+          if (dissimilarity > MAX_DISSIMILARITY) return Infinity
         }
       }
     }
   }
 
-  return ret
+  return dissimilarity
 }
 
 /** return the dissimilarity between two individual transit segments (each with only a single pattern, not yet merged) */
-function segmentDissimilarity (s1, s2, data) {
-  const pat1 = data.patterns.find((p) => p.pattern_id === s1.pattern_id)
-  const pat2 = data.patterns.find((p) => p.pattern_id === s2.pattern_id)
-
-  const from1 = pat1.stops[s1.from_stop_index].stop_id
-  const to1 = pat1.stops[s1.to_stop_index].stop_id
-  const from2 = pat2.stops[s2.from_stop_index].stop_id
-  const to2 = pat2.stops[s2.to_stop_index].stop_id
-
-  const findStop = (id) => data.stops.find((stop) => stop.stop_id === (id + ''))
-  const d1 = stopDistance(findStop(from1), findStop(from2))
-  const d2 = stopDistance(findStop(to1), findStop(to2))
+function segmentDissimilarity (s1: Segment, s2: Segment, {
+  patterns,
+  stops
+}: {
+  patterns: Array<Pattern>,
+  stops: Array<Stop>
+}): number {
+  const pat1 = patterns.find((p) => p.pattern_id === s1.pattern_id)
+  const pat2 = patterns.find((p) => p.pattern_id === s2.pattern_id)
+
+  if (!pat1 || !pat2) return Infinity
```

Comment:
`find` can return `undefined` and `flow` won't let you call a method on a possibly `undefined` value